web/src/components/SecondaryNav.test.js

```diff
@@ -1,9 +1,81 @@
 import { shallow } from 'enzyme';
 import React from 'react';
-import SecondaryNav from './SecondaryNav';
+import {
+  plain as SecondaryNav,
+  mapStateToProps,
+  mapDispatchToProps
+} from './SecondaryNav';
+import { addActivity } from '../actions/editActivity';
+import { jumpTo } from '../actions/app';
+
+global.scrollTo = jest.fn();
 
 describe('Secondary Nav component', () => {
-  it('renders correctly', () => {
-    expect(shallow(<SecondaryNav />).dive()).toMatchSnapshot();
+  const props = {
+    activities: [{ anchor: 'activity-1' }, { anchor: 'activity-2' }],
+    activityCount: 2,
+    add: jest.fn(),
+    jumpTo: jest.fn()
+  };
+
+  beforeEach(() => {
+    props.add.mockClear();
+    props.jumpTo.mockClear();
+  });
+
+  it('renders correctly without add activity button', () => {
+    expect(
+      shallow(<SecondaryNav {...props} activeSection="activity-1-ffp" />).dive()
+    ).toMatchSnapshot();
+  });
```

Comment:
```suggestion
    expect(
      shallow(<SecondaryNav {...props} activeSection="activity-2-something-else"/>).dive()
    ).toMatchSnapshot();
  });
```

Also test the condition that we're on the last activity but *not* on the FFP section.